app/dashboard/models.py

```diff
@@ -400,6 +415,73 @@ def fetch_parameters_values(self):
                     )
                 return simulations_results
 
+        if self.report_type == GRAPH_CAPACITIES:
+            y_variables = parameters.get("y", None)
+            scenario = get_object_or_404(Scenario, pk=self.project_id)
+
+            if y_variables is not None:
+                simulations_results = []
+
+                for simulation in self.simulations.all():
+                    y_values = (
+                        []
+                    )  # stores the capacity, both installed and optimized in seperate dicts, of each individual asset/ component
+                    x_values = []  # stores the label of the corresponding asset
+                    assets_results_obj = AssetsResults.objects.get(
+                        simulation=simulation
+                    )
+
+                    results_dict = json.loads(
+                        Scenario.objects.get(
+                            simulation=scenario.simulation
+                        ).simulation.results
+                    )
+
+                    kpi_scalar_matrix = results_dict["kpi"]["scalar_matrix"]
+
+                    installed_capactiy_dict = {
+                        "capacity": [],
+                        "name": "Installed Capacity",
+                    }
+                    optimized_capacity_dict = {
+                        "capacity": [],
+                        "name": "Optimized Capacity",
+                    }
+
+                    for y_var in y_variables:
+                        x_values.append(
+                            y_var
+                            + " in "
+                            + assets_results_obj.single_asset_results(asset_name=y_var)[
+                                "installed_capacity"
+                            ]["unit"]
```

Comment:
Because you use assets_results_obj.single_asset_results(asset_name=y_var["installed_capacity"] twice you could store it in a variable with a shorter more telling name, which would help code readability. FOr example `installed_cap = assets_results_obj.single_asset_results(asset_name=y_var["installed_capacity"]`, then `installed_cap["unit"]` and `installed_cap["value"]`